test/framework/src/main/java/org/elasticsearch/test/hamcrest/ElasticsearchAssertions.java

```diff
@@ -669,6 +670,11 @@ public static void assertVersionSerializable(Version version, Streamable streama
                     equalTo(0));
             assertThat("Serialization failed with version [" + version + "] bytes should be equal for streamable [" + streamable + "]",
                     serialize(version, streamable), equalTo(orig));
+        } catch (UnsupportedOperationException ex) {
+            if (ex.equals("can't read named writeable from StreamInput")) {
```

Comment:
Oh shit! We can totally rethrow. I'd really like to a way to get the NamedWriteableRegistry there. Maybe we can get it somehow. I just wrote this to work around these "surprise" failures. Part of what makes this a "proposal" is that I'm perfectly happy if we say "we can do this, but not until 6.0 and not until we get NamedWriteableRegistry here".